### PR TITLE
MobX tests

### PR DESCRIPTION
**Pull request by sadasant**
Here are some MobX tests. The tests aren't fully cleaned up yet.

**Notes:**
Reactions only work for pre-defined things. We can use some `treeUtils` to set those things on the fly before triggering updates and so on, but they need to exist. Otherwise, mutation, adding and deleting seem to be working fine.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -97,27 +99,20 @@ export let ContextTree = (
...
-  let {add, remove, mutate} = actions({ getNode, flat, dispatch })
   let subscribe = (f, cond = _.stubTrue) => {
     let index = subscribers.length
     // Potential improvement - optimize for `path` cases and store locally at node (assuming we have lots of subscriptions to different nodes)
     subscribers.push([_.iteratee(cond), f])
     return () => subscribers.splice(index, 1)
   }
 
-  return {
+  return _.extend(actions({ getNode, flat, dispatch, snapshot, extend }), {
```
Comment (daedalus28) [inline src/index.js]: use spread...

File: test/mobx.js [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+          data: {
+            query: 'cable'
+          },
+          filterOnly: true
+        }]
+      }]
+    })
+  })
+
+  it('should subscribe and unsubscribe', async () => {
```
Comment (daedalus28) [inline test/mobx.js]: this has nothing to do with mobx, lets kill it

File: test/mobx.js [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+import Promise from 'bluebird'
+const expect = chai.expect
+chai.use(sinonChai)
+
+let traverse = x => x && x.children && x.children.slice() // mobx needs slice
+let treeUtils = F.tree(traverse, keyPath)
+let ContextTreeMobx = (tree, service) => lib.ContextTree(tree, service, undefined, { snapshot: toJS, extend: extendObservable })
+
+describe('mobx', () => {
+  let tree = observable({
```
Comment (daedalus28) [inline test/mobx.js]: it looks like we're instantiating a couple of trees - we might only need 1 for these tests

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Your PR has lint errors. Please fix these and commit them.
      
<details>
  <summary>mobx.js</summary>
  <p>Full Path: <code>/home/circleci/project/test/mobx.js</code></p>
  
  <strong>'flattenTree' is defined but never used.</strong>
  <code>Line 1: import {flattenTree, keyPath} from '../src/util/tree'</code>

  <strong>'util' is defined but never used.</strong>
  <code>Line 9: import util from 'util'</code>
</details>
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  The changelog has not been updated. Please update the changelog.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  The version was not updated. Please update the version.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  You should add at least 1 more reviewer to the PR
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Your PR has lint warnings. Please consider fixing these.
      
<details>
  <summary>index.js</summary>
  <p>Full Path: <code>/home/circleci/project/src/index.js</code></p>
  
  <strong>Unexpected console statement.</strong>
  <code>Line 47: let log = x => debug && console.log(x)</code>
</details>
<details>
  <summary>reactors.js</summary>
  <p>Full Path: <code>/home/circleci/project/src/reactors.js</code></p>
  
  <strong>Expected '===' and instead saw '=='.</strong>
  <code>Line 9: parent.join == 'or' ? [] : _.difference(parent.children, [instigator]),</code>

  <strong>Expected '===' and instead saw '=='.</strong>
  <code>Line 55: let joinInverted = instigator.join == 'not' || previous.join == 'not'</code>

  <strong>Expected '===' and instead saw '=='.</strong>
  <code>Line 55: let joinInverted = instigator.join == 'not' || previous.join == 'not'</code>
</details>
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  The README has not been updated. Please update the README.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  Could not find any browser results.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  No `duti:fix` npm script found. Not autofixing this PR.
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**